R/theme-elements.r

```diff
@@ -244,144 +244,277 @@ element_grob.element_line <- function(element, x = 0:1, y = 0:1,
+      axis.title.x.bottom   = el_def("element_text", "axis.title.x"),
+      axis.title.y          = el_def("element_text", "axis.title"),
+      axis.title.y.left     = el_def("element_text", "axis.title.y"),
+      axis.title.y.right    = el_def("element_text", "axis.title.y"),
+
+      legend.background     = el_def("element_rect", "rect"),
+      legend.margin         = el_def("margin"),
+      legend.spacing        = el_def("unit"),
+      legend.spacing.x      = el_def("unit", "legend.spacing"),
+      legend.spacing.y      = el_def("unit", "legend.spacing"),
+      legend.key            = el_def("element_rect", "rect"),
+      legend.key.height     = el_def("unit", "legend.key.size"),
+      legend.key.width      = el_def("unit", "legend.key.size"),
+      legend.text           = el_def("element_text", "text"),
+      legend.text.align     = el_def("character"),
+      legend.title          = el_def("element_text", "title"),
+      legend.title.align    = el_def("character"),
+      legend.position       = el_def("character"),  # Need to also accept numbers
+      legend.direction      = el_def("character"),
+      legend.justification  = el_def("character"),
+      legend.box            = el_def("character"),
+      legend.box.just       = el_def("character"),
+      legend.box.margin     = el_def("margin"),
+      legend.box.background = el_def("element_rect", "rect"),
+      legend.box.spacing    = el_def("unit"),
+
+      panel.background      = el_def("element_rect", "rect"),
+      panel.border          = el_def("element_rect", "rect"),
+      panel.spacing         = el_def("unit"),
+      panel.spacing.x       = el_def("unit", "panel.spacing"),
+      panel.spacing.y       = el_def("unit", "panel.spacing"),
+      panel.grid.major.x    = el_def("element_line", "panel.grid.major"),
+      panel.grid.major.y    = el_def("element_line", "panel.grid.major"),
+      panel.grid.minor.x    = el_def("element_line", "panel.grid.minor"),
+      panel.grid.minor.y    = el_def("element_line", "panel.grid.minor"),
+      panel.ontop           = el_def("logical"),
+
+      strip.background      = el_def("element_rect", "rect"),
+      strip.text.x          = el_def("element_text", "strip.text"),
+      strip.text.y          = el_def("element_text", "strip.text"),
+      strip.placement       = el_def("character"),
+      strip.placement.x     = el_def("character", "strip.placement"),
+      strip.placement.y     = el_def("character", "strip.placement"),
+      strip.switch.pad.grid = el_def("unit"),
+      strip.switch.pad.wrap = el_def("unit"),
+
+      plot.background       = el_def("element_rect", "rect"),
+      plot.title            = el_def("element_text", "title"),
+      plot.subtitle         = el_def("element_text", "title"),
+      plot.caption          = el_def("element_text", "title"),
+      plot.margin           = el_def("margin"),
+
+      aspect.ratio          = el_def("character")
+    ),
```

Comment:
varElementsFixedDefault: 
This is the default tree structure under ggplot2. Previously this was .element_tree